From 9bb2f4021a5aa3a876fb8028bd8db85054d73d12 Mon Sep 17 00:00:00 2001
From: Juri Lelli <juri.lelli@arm.com>
Date: Fri, 6 Mar 2015 11:01:28 +0000
Subject: [PATCH 28/38] DEBUG: add cpu to load_avg_task tracepoint

This is useful when plotting where a task was running during an
experiment.

Change-Id: I5a90aaf3eb1254b4d1c6ec3d7e4159d9643b0b71
Signed-off-by: Juri Lelli <juri.lelli@arm.com>
---
 include/trace/events/sched.h | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/include/trace/events/sched.h b/include/trace/events/sched.h
index f998e4b..5ce115d 100644
--- a/include/trace/events/sched.h
+++ b/include/trace/events/sched.h
@@ -591,6 +591,7 @@ TRACE_EVENT(sched_load_avg_task,
 	TP_STRUCT__entry(
 		__array( char,	comm,	TASK_COMM_LEN		)
 		__field( pid_t,	pid				)
+		__field( int,	cpu				)
 		__field( unsigned long,	load			)
 		__field( unsigned long,	utilization		)
 		__field( unsigned int,	runnable_avg_sum	)
@@ -601,6 +602,7 @@ TRACE_EVENT(sched_load_avg_task,
 	TP_fast_assign(
 		memcpy(__entry->comm, tsk->comm, TASK_COMM_LEN);
 		__entry->pid			= tsk->pid;
+		__entry->cpu			= task_cpu(tsk);
 		__entry->load			= avg->load_avg_contrib;
 		__entry->utilization		= avg->utilization_avg_contrib;
 		__entry->runnable_avg_sum	= avg->runnable_avg_sum;
@@ -608,9 +610,10 @@ TRACE_EVENT(sched_load_avg_task,
 		__entry->avg_period		= avg->avg_period;
 	),
 
-	TP_printk("comm=%s pid=%d load=%lu utilization=%lu runnable_avg_sum=%u"
+	TP_printk("comm=%s pid=%d cpu=%d load=%lu utilization=%lu runnable_avg_sum=%u"
 		  " running_avg_sum=%u avg_period=%u",
-		  __entry->comm, __entry->pid, __entry->load, __entry->utilization,
+		  __entry->comm, __entry->pid, __entry->cpu,
+		  __entry->load, __entry->utilization,
 		  (unsigned int)__entry->runnable_avg_sum,
 		  (unsigned int)__entry->running_avg_sum,
 		  (unsigned int)__entry->avg_period)
-- 
1.9.1

